client/src/components/mapping/Comments/index.tsx

```diff
@@ -22,78 +24,130 @@ const Comments = () => {
   const { attribute, resource } = useSelector(
     (state: IReduxStore) => state.selectedNode
   );
+  const me = useSelector((state: IReduxStore) => state.user);
 
   const attributesForResource = useSelector(
     (state: IReduxStore) => state.resourceInputs.attributesMap
   );
   const attributeForNode = attributesForResource[attribute.path];
 
   const [createAttribute] = useMutation(mCreateAttribute);
-  const [updateComments] = useMutation(mUpdateComments);
+  const [createComment] = useMutation(mCreateComment);
 
-  const [comments, setComments] = React.useState(
-    attributeForNode ? attributeForNode.comments : ''
-  );
+  const [newComment, setNewComment] = React.useState('');
+  const [comments, setComments] = React.useState([] as IComment[]);
 
-  const { data, loading } = useQuery(qCommentsForAttribute, {
+  const { data: attrWithComments, loading } = useQuery(qCommentsForAttribute, {
     variables: {
       attributeId: attributeForNode ? attributeForNode.id : null
     },
     skip: !attributeForNode
   });
 
   React.useEffect(() => {
-    setComments(data && data.attribute.comments ? data.attribute.comments : '');
-  }, [data]);
+    if (attrWithComments)
+      setComments(
+        attrWithComments.attribute.comments.sort(
+          (c1: IComment, c2: IComment) =>
+            new Date(c2.createdAt).getTime() - new Date(c1.createdAt).getTime()
+        )
+      );
+  }, [attrWithComments]);
+
+  React.useEffect(() => {
+    if (!attributeForNode) setComments([]);
+  }, [attributeForNode]);
+
+  const onCreateComment = (
+    cache: any,
+    { data: { createComment: newComment } }: any
+  ) => {
+    const { attribute } = cache.readQuery({
+      query: qCommentsForAttribute,
+      variables: {
+        attributeId: attributeForNode.id
+      }
+    });
+    cache.writeQuery({
+      query: qCommentsForAttribute,
+      variables: {
+        attributeId: attributeForNode.id
+      },
+      data: {
+        attribute: {
+          ...attribute,
+          comments: [...attribute.comments, newComment]
+        }
+      }
+    });
+  };
 
-  const onSaveComment = async (): Promise<void> => {
+  const actionCreateComment = async (): Promise<void> => {
+    let attributeId = attributeForNode?.id;
     try {
-      if (attributeForNode) {
-        updateComments({
-          variables: {
-            attributeId: attributeForNode.id,
-            comments
-          }
-        });
-      } else {
+      if (!attributeForNode) {
         const { data } = await createAttribute({
           variables: {
             resourceId: resource.id,
-            path: attribute.path,
-            data: {
-              comments
-            }
+            definitionId: attribute.types[0],
+            path: attribute.path
           }
         });
         const newAttr = data.createAttribute;
+        attributeId = newAttr.id;
```

Comment:
I don't know why this was in the other components but it shouldn't be here!

Reply:
yes it should, I need it in the mutation just below that ?

Reply:
Oops, misread!
You don't create the parents in the DB when you add a comment?